### PR TITLE
Editors: validation message position is not correct if a parent container was hidden (T1095900)

### PR DESCRIPTION
**Pull request by EugeniyKiyashko**

### CODE REVIEW

File: testing/testcafe/tests/editors/validationMessage.ts [hunk 1]
```diff
@@ -20,3 +27,55 @@ test('ValidationMessage integrated in editor should not raise any errors when it
...
+    .ok()
+    .expect(compareResults.isValid())
+    .ok(compareResults.errorMessages());
+
+  await ClientFunction(() => {
+    (document.querySelector('#container') as HTMLElement).setAttribute('hidden', 'true');
+  })();
+
+  await ClientFunction(() => {
+    document.querySelector('#container')?.removeAttribute('hidden');
```
Comment (ksercs) [inline testing/testcafe/tests/editors/validationMessage.ts:53]: cannot we do this in the previous ClientFunction?

File: testing/testcafe/tests/editors/validationMessage.ts [hunk 2]
```diff
@@ -20,3 +27,55 @@ test('ValidationMessage integrated in editor should not raise any errors when it
     .click(checkBox1.element)
     .expect(true).ok();
 });
+
+fixture`ValidationMessage_TextBox`
+  .page(url(__dirname, '../container.html'));
+
+test('Validation Message position should be correct after change visibility of parent container', async (t) => {
```
Comment (ksercs) [inline testing/testcafe/tests/editors/validationMessage.ts]:
```suggestion
test('Validation Message position should be correct after change visibility of parent container (T1095900)', async (t) => {
```

File: testing/testcafe/tests/editors/validationMessage.ts [hunk 3]
```diff
@@ -20,3 +27,55 @@ test('ValidationMessage integrated in editor should not raise any errors when it
...
+    document.querySelector('#container')?.removeAttribute('hidden');
+  })();
+
+  await t
+    .expect(await takeScreenshot('Textbox_validation_message.png'))
+    .ok()
+    .expect(compareResults.isValid())
+    .ok(compareResults.errorMessages());
+}).before(async (t) => {
+  await t.resizeWindow(300, 200);
```
Comment (ksercs) [inline testing/testcafe/tests/editors/validationMessage.ts:62]: why do we need this window resize?

File: testing/testcafe/tests/editors/validationMessage.ts [hunk 4]
```diff
@@ -1,9 +1,16 @@
...
+import { restoreBrowserSize } from '../../helpers/restoreBrowserSize';
+import createWidget from '../../helpers/createWidget';
+import { appendElementTo } from '../navigation/helpers/domUtils';
+
+const TEXTEDITOR_INPUT_CLASS = 'dx-texteditor-input';
 
 fixture`ValidationMessage`
   .page(url(__dirname, './pages/T941581.html'))
-  .beforeEach(async (t) => { await t.wait(10000); });
+  .beforeEach(async (t) => { await t.wait(1000); });
```
Comment (ksercs) [inline testing/testcafe/tests/editors/validationMessage.ts]: why is it necessary?

File: js/ui/overlay/ui.overlay.js [hunk 5]
```diff
@@ -334,7 +334,7 @@ const Overlay = Widget.inherit({
     },
 
     _contentResizeHandler: function(entry) {
-        if(!this._shouldSkipContentResize(entry)) {
+        if(!this._shouldSkipContentResize(entry) && this._$content.is(':visible')) {
```
Comment (ksercs) [inline js/ui/overlay/ui.overlay.js:337]: Isn't it better to use our api to check if popup is opened?

File: testing/testcafe/tests/editors/validationMessage.ts (hunk 1 above)
Comment (EugeniyKiyashko) [reply testing/testcafe/tests/editors/validationMessage.ts:53]: I do setAttribute and removeAttribute logic in various clientFunction willfully. It helps me separate important parts of scenario logic separate each other and debug it.

File: testing/testcafe/tests/editors/validationMessage.ts (hunk 3 above)
Comment (EugeniyKiyashko) [reply testing/testcafe/tests/editors/validationMessage.ts:62]: I need to do screenshot which contains the Texteditor and the ValidationMessage simultaneously. There is no container element that include both of them. So, I get minimal screenshot dimensions for my purpose through resize windows.

File: js/ui/overlay/ui.overlay.js (hunk 5 above)
Comment (EugeniyKiyashko) [reply js/ui/overlay/ui.overlay.js:337]: No, the parent container is hiding. So, visibilityChange for nested component doesn't call and ValidationMessage stay visible. It's important check visibility this way.

**Review by ksercs — APPROVED** [review]

**Pull request merged**